lambdas/letter-stream-forwarder/src/letter-stream-forwarder.ts

```diff
@@ -0,0 +1,58 @@
+import { LetterSchemaBase } from "@internal/datastore";
+import { DynamoDBStreamEvent, DynamoDBRecord, DynamoDBStreamHandler } from "aws-lambda";
+import { PutRecordCommand } from "@aws-sdk/client-kinesis";
+import { Deps } from "./deps";
+import { unmarshall } from "@aws-sdk/util-dynamodb";
+import { LetterStatus } from "../../api-handler/src/contracts/letters";
+import { Logger } from "pino";
+
+const VALID_STATE_TRANSITIONS: Record<LetterStatus, Set<LetterStatus>> = {
```

Comment:
My worry here is that if something goes wrong we are excluding the possibility of updating the state from an incorrect terminal state.

I think this could make sense to filter when accepting state transitions from the supplier, but filtering out our own internal state transitions after they have been accepted and stored in the DB seems like we may hide information that the rest of the service may want.

For example, if something transitions from `FAILED` to `DISPATCHED` and we accept that transition in the API, we may not correctly invoice for something that will be charged by the supplier as a successful delivery, and the upstream services will not have the correct status.